x-pack/plugins/elastic_assistant/server/lib/langchain/elasticsearch_store/elasticsearch_store.ts

```diff
@@ -0,0 +1,316 @@
+   */
+  addDocuments = async (
+    documents: Document[],
+    options?: Record<string, never>
+  ): Promise<string[] | void> => {
+    const pipelineExists = await this.pipelineExists();
+    if (!pipelineExists) {
+      await this.createPipeline();
+    }
+    const indexExists = await this.indexExists();
+    if (!indexExists) {
+      await this.createIndex();
+    }
+
+    const operations = documents.flatMap(({ pageContent, metadata }) => [
+      { index: { _index: this.index, _id: uuid.v4() } },
+      { text: pageContent, metadata },
+    ]);
+
+    try {
+      const response = await this.esClient.bulk({ refresh: true, operations });
+      return response.errors
+        ? []
+        : response.items.flatMap((item) => (item.create?._id ? [item.create?._id] : []));
+    } catch (e) {
+      this.logger.error('Error loading data into KB', e);
+    }
+  };
+
+  /**
+   * Add vectors to the store. Returns a list of document IDs.
+   *
+   * @param vectors Vector representation of documents to add to the store
+   * @param documents Documents corresponding to the provided vectors
+   * @param options Any additional options as defined in the interface
+   * @returns Promise<string[]> of document IDs added to the store
+   */
+  addVectors = (
+    vectors: number[][],
+    documents: Document[],
+    options?: {}
+  ): Promise<string[] | void> => {
+    // Note: implement if/when needed
+    return Promise.resolve(undefined);
+  };
+
+  /**
+   * Performs similarity search on the store using the provided query vector and filter, returning k similar
+   * documents along with their score.
+   *
+   * @param query Query vector to search with
+   * @param k Number of similar documents to return
+   * @param filter Optional filter to apply to the search
+   *
+   *  @returns Promise<Array<[Document, number]>> of similar documents and their scores
+   */
+  similaritySearchVectorWithScore = (
+    query: number[],
+    k: number,
+    filter?: this['FilterType']
+  ): Promise<Array<[Document, number]>> => {
+    // Note: Implement if needed
+    return Promise.resolve([]);
+  };
+
+  // Non-abstract function overrides
+
+  /**
+   * Performs similarity search on the store using the provided query string and filter, returning k similar
+   * @param query Query vector to search with
+   * @param k Number of similar documents to return
+   * @param filter Optional filter to apply to the search
+   * @param _callbacks Optional callbacks
+   *
+   * @returns Promise<Document[]> of similar documents
+   */
+  similaritySearch = async (
+    query: string,
+    k?: number,
+    filter?: this['FilterType'] | undefined,
+    _callbacks?: Callbacks | undefined
+  ): Promise<Document[]> => {
+    const queryBody: QueryDslQueryContainer = {
+      bool: {
+        must: [
+          {
+            text_expansion: {
+              'vector.tokens': {
+                model_id: '.elser_model_1',
```

Comment:
consider importing this constant (or making it a variable)

Reply:
Hoisted to class state and included as optional param in constructor ala `index`. Initialized to `this.model = model ?? 'elser_model_1';`